File: src/main/java/thedarkcolour/futuremc/item/HoneyBottleItem.java
package thedarkcolour.futuremc.item;

import net.minecraft.advancements.CriteriaTriggers;
import net.minecraft.entity.LivingEntity;
import net.minecraft.entity.player.PlayerEntity;
import net.minecraft.entity.player.ServerPlayerEntity;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.item.UseAction;
import net.minecraft.potion.Effects;
import net.minecraft.stats.Stats;
import net.minecraft.util.ActionResult;
import net.minecraft.util.ActionResultType;
import net.minecraft.util.Hand;
import net.minecraft.world.World;
import net.minecraftforge.event.entity.PlaySoundAtEntityEvent;
import thedarkcolour.futuremc.sound.Sounds;

public class HoneyBottleItem extends Item {
    public HoneyBottleItem() {
        super(new Item.Properties().maxStackSize(1));
        setRegistryName("honey_bottle");
    }

    @Override
    public UseAction getUseAction(ItemStack stack) {
        return UseAction.EAT;
    }

    @Override
    public int getUseDuration(ItemStack stack) {
        return 40;
    }

    @Override
    public ActionResult<ItemStack> onItemRightClick(World worldIn, PlayerEntity playerIn, Hand handIn) {
        playerIn.setActiveHand(handIn);
        return new ActionResult<>(ActionResultType.SUCCESS, playerIn.getHeldItem(handIn));
    }

    @Override
    public ItemStack onItemUseFinish(ItemStack stack, World worldIn, LivingEntity entityLiving) {
        super.onItemUseFinish(stack, worldIn, entityLiving);
        if (entityLiving instanceof ServerPlayerEntity) {
            ServerPlayerEntity serverPlayer = (ServerPlayerEntity) entityLiving;
            CriteriaTriggers.CONSUME_ITEM.trigger(serverPlayer, stack);
            serverPlayer.addStat(Stats.ITEM_USED.get(this));
        }

        if (!worldIn.isRemote) {
            entityLiving.removeActivePotionEffect(Effects.POISON);
        }

        return stack.isEmpty() ? new ItemStack(Items.GLASS_BOTTLE) : stack;
    }

    @Override
    public ItemStack getContainerItem(ItemStack itemStack) {
        return new ItemStack(Items.GLASS_BOTTLE);
    }

    public static void onHoneyBottleEaten(final PlaySoundAtEntityEvent event) {
        if (event.getEntity() instanceof LivingEntity) {
            LivingEntity e = (LivingEntity) event.getEntity();
            if (e.getActiveItemStack().getItem() == Items.HONEY_BOTTLE) {
                event.setSound(Sounds.HONEY_BOTTLE_DRINK);
            }
        }
    }
}

File: src/main/java/thedarkcolour/futuremc/item/HoneyBottleItem.kt
package thedarkcolour.futuremc.item

import net.minecraft.advancements.CriteriaTriggers
import net.minecraft.creativetab.CreativeTabs
import net.minecraft.entity.EntityLivingBase
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.entity.player.EntityPlayerMP
import net.minecraft.init.Items
import net.minecraft.init.MobEffects
import net.minecraft.item.EnumAction
import net.minecraft.item.ItemFood
import net.minecraft.item.ItemStack
import net.minecraft.stats.StatList
import net.minecraft.util.ActionResult
import net.minecraft.util.EnumActionResult
import net.minecraft.util.EnumHand
import net.minecraft.world.World
import thedarkcolour.core.util.setItemModel
import thedarkcolour.core.util.setItemName
import thedarkcolour.futuremc.FutureMC
import thedarkcolour.futuremc.config.FConfig

// work with apple skin and quark
class HoneyBottleItem : ItemFood(6, 0.8f, false) {
    init {
        // use functions from ItemModeled.kt
        setItemName(this, "honey_bottle")
        setItemModel(this, 0)
        setMaxStackSize(16)
        creativeTab = if (FConfig.useVanillaCreativeTabs) CreativeTabs.FOOD else FutureMC.GROUP
        containerItem = Items.GLASS_BOTTLE
    }

    override fun getItemUseAction(stack: ItemStack) = EnumAction.DRINK

    override fun getMaxItemUseDuration(stack: ItemStack) = 40

    override fun onItemRightClick(worldIn: World, playerIn: EntityPlayer, handIn: EnumHand): ActionResult<ItemStack> {
        playerIn.activeHand = handIn
        return ActionResult(EnumActionResult.SUCCESS, playerIn.getHeldItem(handIn))
    }

    override fun onItemUseFinish(stack: ItemStack, worldIn: World, entityLiving: EntityLivingBase): ItemStack {
        super.onItemUseFinish(stack, worldIn, entityLiving)
        if (entityLiving is EntityPlayerMP) {
            CriteriaTriggers.CONSUME_ITEM.trigger(entityLiving, stack)
            entityLiving.addStat(StatList.getObjectUseStats(this)!!)
        }

        if (!worldIn.isRemote) {
            entityLiving.removePotionEffect(MobEffects.POISON)
        }

        return if (stack.isEmpty) {
            ItemStack(Items.GLASS_BOTTLE)
        } else {
            val bottle = ItemStack(Items.GLASS_BOTTLE)

            if (entityLiving is EntityPlayer && !entityLiving.capabilities.isCreativeMode) {
                if (!entityLiving.inventory.addItemStackToInventory(bottle)) {
                    entityLiving.dropItem(bottle, false)
                }
            }

            stack
        }
    }
}